src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -799,7 +823,7 @@ private static void ParseStatusLine(Span<byte> line, HttpResponseMessage respons
             const int MinStatusLineLength = 12; // "HTTP/1.x 123" 
             if (line.Length < MinStatusLineLength || line[8] != ' ')
             {
-                ThrowInvalidHttpResponse();
+                TryThrowInvalidHttpResponse(isFromTrailer:false);
```

Comment:
Why are all of these changing?  They should continue to just call ThrowInvalidHttpResponse as they did previously.  If there are places where ThrowInvalidHttpResponse should only be called conditionally, you can just check whether it's from a trailer at the call site or add an additional helper to do that check if necessary.

Reply:
It seems like we should still throw in all of these cases, and stop processing and kill the connection, but also just catch the exception in the calling code since there's no way to report this error to users.

Reply:
> just catch the exception in the calling code since there's no way to report this error to users.

Not while draining, but during normal trailer processing the exception should just naturally come out of the ReadAsync task. And presumably for draining we already have a catch somewhere, since excepting could already occur for other reasons whole draining...?

Reply:
I was thinking we should be more tolerable (skip, not throw) for invalid trailing headers.

Will change to throwing to follow `Headers` pattern.

Reply:
Hmm, that could be considered a regression. Content that was previously ignored can now cause an exception at the end of the request?

Reply:
I'm also confused here, but my interpretation is: throw for invalid headers - to kill the connection, then catch the exception in the calling code (i.e., in `ReadChunkFromConnectionBuffer()`).